client/controller/shared.js

```diff
@@ -35,25 +37,86 @@ export function makeLayoutMiddleware( LayoutComponent ) {
 	};
 }
 
+/**
+ * Loads a a css stylesheet into the page
+ * @param {string} cssUrl - a url to a css resource to be inserted into the page
+ * @param {Function} callback - a callback function to be called when the CSS has been loaded (after 500ms have passed).
+ */
+function loadCSS( cssUrl, callback = noop ) {
+	const link = Object.assign( document.createElement( 'link' ), {
+		rel: 'stylesheet',
+		type: 'text/css',
+		href: cssUrl,
+	} );
+
+	const onload = () => {
+		if ( 'onload' in link ) {
+			link.onload = null;
+		}
+		callback( null, link );
+	};
+
+	if ( 'onload' in link ) {
+		link.onload = onload;
+	} else {
+		setTimeout( onload, 500 );
+	}
+
+	document.head.appendChild( link );
+}
+
 export function setSection( section ) {
 	return ( context, next = noop ) => {
 		context.store.dispatch( setSectionAction( section ) );
 
-		next();
+		loadSectionCSS( context, next );
 	};
 }
 
+export function loadSectionCSS( context, next ) {
+	const section = getSection( context.store.getState() );
+
+	if ( section.cssUrls && typeof document !== 'undefined' ) {
+		const cssUrl = context.isRTL ? section.cssUrls.rtl : section.cssUrls.ltr;
+
+		// TODO: handle adding styles in `state.documentHead.meta` instead (currently only supports setting all meta at once)
+		const currentLink = document.getElementById( 'section-css' );
+
+		// if the url has not changed, skip
+		if ( currentLink && currentLink.getAttribute( 'href' ) === cssUrl ) {
+			return next();
+		}
+
+		loadCSS( cssUrl, ( err, newLink ) => {
+			// remove old css if it exists and is still in the dom
+			if ( currentLink && currentLink.parentElement ) {
+				currentLink.parentElement.removeChild( currentLink );
```

Comment:
Why did we not use [`currentLink.remove()`](https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/remove) instead here?

Reply:
IE doesn't support remove() but supports [https://msdn.microsoft.com/en-us/library/ms536702(v=vs.85).aspx]( removeChild() )